tools/archives/pyvo_integration/astronomical_archives.xml

```diff
@@ -0,0 +1,1023 @@
+
+          #end if
+
+        #end if
+
+        python '$__tool_directory__/astronomical_archives.py' '$output' '$output_csv' '$output_html' '$output_basic_html' '$output_error'
+      ]]>
+    </command>
+    <inputs>
+        <conditional name="archiveselection">
+            <param name="archive_type" type="select" label="Archive Selection">
+              <option value="registry">Query IVOA registry</option>
+              <option value="archive">Query specific IVOA archive</option>
+            </param>
+            <when value="registry">
+              <param name="keyword" type="text" label="Keyword" />
+              <param name="service_type" type="select" label="Service type">
+                <option value="TAP" selected="true">TAP: Tables</option>
+                <option value="SCS">SCS: Cone search</option>
+                <option value="SSA">SSA: Spectra</option>
+                <option value="SIA">SIA: Images</option>
+              </param>
+              <param name="wavebands" type="select" label="Wavebands">
+                <option value="all" selected="true">All</option>
+                <option value="radio">Radio</option>
+                <option value="millimeter">Millimeter</option>
+                <option value="infrared">Infrared</option>
+                <option value="optical">Optical</option>
+                <option value="uv">UV</option>
+                <option value="euv">EUV</option>
+                <option value="x-ray">X-ray</option>
+                <option value="gamma-ray">Gamma-ray</option>
+              </param>
+            </when>
+            <when value="archive">
+              <param name="archive" type="select" label="Archive">
+                <options from_data_table="astronomical_archives" />
+              </param>
+            </when>
+        </conditional>
+        <section name="query_section" title="Query selection" expanded="false">
+          <conditional name="query_selection">
+            <param name="query_type" type="select" label="ADQL Query Selection">
+              <option value="none">No specific query (first n files from archive)</option>
+              <option value="obscore_query">IVOA obscore table query builder</option>
+              <option value="raw_query">Raw ADQL query builder</option>
+            </param>
+            <when value="none"></when>
+            <when value="obscore_query">
+              <!-- Fields defined https://www.ivoa.net/documents/ObsCore/20160224/WD-ObsCore-v1.1-20160224.pdf -->
+              <param name="target_name" type="text" label="Observation target name" help="Typically name of the astronomical object observed" />
+              <param name="t_min" type="float" optional="true" label="Start time in MJD" />
```

Comment:
can we add min/max?

Reply:
Min is a bit arbitrary (0 is fine in most cases, it's 1858), and max should in principle depend on current moment, but I set now+10 years. We can not make it dynamic I guess?

```suggestion
              <param name="t_min" type="float" min=0 max=63832 optional="true" label="Start time in MJD" />
```

Reply:
dynamic? you mean dependent on another param? No that is not possible yet. If no min/max possible, then leave it like it is. min/max can just improve UX

Reply:
It's current moment, so depending on current time.